src/main/java/org/mule/extension/sftp/internal/connection/SftpClient.java

```diff
@@ -444,17 +445,21 @@ public String getHome() {
    * The solution at the moment is to add a timeout with the tools provided by mule sdk.
    */
   private String executePWDCommandWithTimeout() throws IOException {
+    Future<String> future = null;
     try {
       Scheduler getHomeScheduler =
           schedulerService.cpuLightScheduler(SchedulerConfig.config().withShutdownTimeout(PWD_COMMAND_EXECUTION_TIMEOUT,
                                                                                           PWD_COMMAND_EXECUTION_TIMEOUT_UNIT));
-      Future<String> submit = getHomeScheduler.submit(() -> session.executeRemoteCommand(PWD_COMMAND));
-      return submit.get().trim();
+      future = getHomeScheduler.submit(() -> session.executeRemoteCommand(PWD_COMMAND));
+      return future.get(PWD_COMMAND_EXECUTION_TIMEOUT, PWD_COMMAND_EXECUTION_TIMEOUT_UNIT).trim();
     } catch (InterruptedException e) {
       throw new MuleRuntimeException(e);
-    } catch (Exception ex) {
+    } catch (TimeoutException e) {
+      future.cancel(true);
```

Comment:
[suggestion] Should we have a null check for future object here?

Reply:
TimeOutException is only thrown by future.get(..), The code above it does not throw any TimeOutExceoption, other remianing Exceptions are already being handled by the Catch(Exception ex) block.